common/djangoapps/util/models.py

```diff
@@ -43,38 +41,27 @@ def uncompress_string(s):
 
 
 class CompressedTextField(models.TextField):
-    """transparently compress data before hitting the db and uncompress after fetching"""
+    """ TextField that transparently compresses data when saving the the database, and decompresses the data
+    when retrieving it from the database. """
+
+    COMPRESSED_ID = 'compressed'
+    __metaclass__ = models.SubfieldBase
 
     def get_prep_value(self, value):
+        """ Compress the text data. """
         if value is not None:
             if isinstance(value, unicode):
                 value = value.encode('utf8')
             value = compress_string(value)
-            value = value.encode('base64').decode('utf8')
+            value = self.COMPRESSED_ID + value.encode('base64').decode('utf8')
```

Comment:
I'm open to suggestions regarding how to determine if the value is already compressed.